src/solver/optimisation/opt_rename_problem.cpp

```diff
@@ -0,0 +1,396 @@
+#include "opt_rename_problem.h"
+#include <sstream>
+
+using namespace Antares::Data::Enum;
+
+void VariableNamer::SetLinkVariableName(int var, ExportStructDict structDict)
+{
+    auto nvars = currentAssetsStorage.problem->NombreDeVariables;
+    if (nvars > var && currentAssetsStorage.problem->NomDesVariables[var].empty())
+    {
+        const auto location
+          = currentAssetsStorage.origin + AREA_SEP + currentAssetsStorage.destination;
+        auto fullName = BuildName(
+          toString(structDict),
+          LocationIdentifier(location, ExportStructLocationDict::link),
+          TimeIdentifier(currentAssetsStorage.timeStep, ExportStructTimeStepDict::hour));
+        currentAssetsStorage.problem->NomDesVariables[var] = fullName;
+    }
+}
+
+void VariableNamer::SetAreaVariableName(int var, ExportStructDict structDict)
+{
+    auto nvars = currentAssetsStorage.problem->NombreDeVariables;
+    if (nvars > var)
+    {
+        auto fullName = BuildName(
+          toString(structDict),
+          LocationIdentifier(currentAssetsStorage.area, ExportStructLocationDict::area),
+          TimeIdentifier(currentAssetsStorage.timeStep, ExportStructTimeStepDict::hour));
+        currentAssetsStorage.problem->NomDesVariables[var] = fullName;
+    }
+}
+
+void VariableNamer::SetAreaVariableName(int var, ExportStructDict structDict, int layerIndex)
+{
+    auto nvars = currentAssetsStorage.problem->NombreDeVariables;
+    if (nvars > var)
+    {
+        auto fullName = BuildName(
+          toString(structDict),
+          LocationIdentifier(currentAssetsStorage.area, ExportStructLocationDict::area) + SEPARATOR
+            + "Layer<" + std::to_string(layerIndex) + ">",
+          TimeIdentifier(currentAssetsStorage.timeStep, ExportStructTimeStepDict::hour));
+        currentAssetsStorage.problem->NomDesVariables[var] = fullName;
+    }
+}
+
+void VariableNamer::SetThermalClusterVariableName(int var,
```

Comment:
About **SetThermalClusterVariableName** and **SetShortTermStorageVariableName** : these 2 functions seems very similar. What is the difference ? could we not have only one function instead of 2 ?